lib/core/src/recover/model.rs

```diff
@@ -201,14 +202,20 @@ impl RecoveredOnchainDataChainReceive {
     pub(crate) fn derive_partial_state(
         &self,
         expected_user_lockup_amount_sat: Option<u64>,
+        swap_limits: Option<PairLimits>,
         is_expired: bool,
         is_waiting_fee_acceptance: bool,
     ) -> Option<PaymentState> {
+        let unexpected_amount =
+            expected_user_lockup_amount_sat.is_some_and(|expected_lockup_amount_sat| {
+                expected_lockup_amount_sat != self.btc_user_lockup_amount_sat
+            });
+        let amount_out_of_bounds = swap_limits.is_some_and(|limits| {
```

Comment:
Question: In the case the user sends bellow the limit, we don't get an update from boltz in the status update stream?
This logic here seems correct to me, just verifying we don't miss any case from the status stream.

Reply:
We do get one. We get the LockupFailed (same event as the happy amountless case), but then we fail to handle it as amountless, so we consider it refundable.